<a href="https://colab.research.google.com/github/exdsgift/Data-Science-Project-UNIVR/blob/main/Fatalities_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**packages:** Importing the most useful packages to perform data exploration and visualization

In [17]:
import pandas as pd
import os
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

# **PART 1:** dataset import and Data Cleaning

The dataset is available at the following link: https://www.kaggle.com/datasets/willianoliveiragibin/fatalities-in-the-israeli-palestinian. It reports data on victims of terrorism in Israel and the war in Palestine.

This notebook aims to analyse events and fatalities by

- Year, month, and day of the month of events that led to fatalities
- Victim profiles such as age, gender, citizenship, participation in hostilities etc
- Event locations, location districts, and location regions
- Type of injury, type of ammunition, and party responsible for killings, among others.

## Upload and first check at the data

In [6]:
fatalities_df = pd.read_csv('drive/MyDrive/fatalities_isr_pse_conflict_2000_to_2023.csv')
fatalities_df.head()

name date_of_event   age  \
0  'Abd a-Rahman Suleiman Muhammad Abu Daghash    2023-09-24  32.0   
1         Usayed Farhan Muhammad 'Ali Abu 'Ali    2023-09-24  21.0   
2             'Abdallah 'Imad Sa'ed Abu Hassan    2023-09-22  16.0   
3             Durgham Muhammad Yihya al-Akhras    2023-09-20  19.0   
4                 Raafat 'Omar Ahmad Khamaisah    2023-09-19  15.0   

   citizenship     event_location event_location_district  \
0  Palestinian     Nur Shams R.C.                 Tulkarm   
1  Palestinian     Nur Shams R.C.                 Tulkarm   
2  Palestinian           Kfar Dan                   Jenin   
3  Palestinian  'Aqbat Jaber R.C.                 Jericho   
4  Palestinian         Jenin R.C.                   Jenin   

  event_location_region date_of_death gender took_part_in_the_hostilities  \
0             West Bank    2023-09-24      M                          NaN   
1             West Bank    2023-09-24      M                          NaN   
2             West Bank    2023-09-22      M                          NaN   
3             West Bank    2023-09-20      M                          NaN   
4             West Bank    2023-09-19      M                          NaN   

  place_of_residence place_of_residence_district type_of_injury  \
0     Nur Shams R.C.                     Tulkarm        gunfire   
1     Nur Shams R.C.                     Tulkarm        gunfire   
2           al-Yamun                       Jenin        gunfire   
3  'Aqbat Jaber R.C.                     Jericho        gunfire   
4              Jenin                       Jenin        gunfire   

        ammunition                killed_by  \
0  live ammunition  Israeli security forces   
1  live ammunition  Israeli security forces   
2  live ammunition  Israeli security forces   
3  live ammunition  Israeli security forces   
4  live ammunition  Israeli security forces   

                                               notes  
0  Fatally shot by Israeli forces while standing ...  
1  Fatally shot by Israeli forces while trying to...  
2  Fatally shot by soldiers while firing at them ...  
3  Shot in the head by Israeli forces while throw...  
4  Wounded by soldiers’ gunfire after running awa...

In [21]:
fatalities_df.head(15).T

0   \
name                                'Abd a-Rahman Suleiman Muhammad Abu Daghash   
date_of_event                                                        2023-09-24   
age                                                                        32.0   
citizenship                                                         Palestinian   
event_location                                                   Nur Shams R.C.   
event_location_district                                                 Tulkarm   
event_location_region                                                 West Bank   
date_of_death                                                        2023-09-24   
gender                                                                        M   
took_part_in_the_hostilities                                                NaN   
place_of_residence                                               Nur Shams R.C.   
place_of_residence_district                                             Tulkarm   
type_of_injury                                                          gunfire   
ammunition                                                      live ammunition   
killed_by                                               Israeli security forces   
notes                         Fatally shot by Israeli forces while standing ...   

                                                                             1   \
name                                       Usayed Farhan Muhammad 'Ali Abu 'Ali   
date_of_event                                                        2023-09-24   
age                                                                        21.0   
citizenship                                                         Palestinian   
event_location                                                   Nur Shams R.C.   
event_location_district                                                 Tulkarm   
event_location_region                                                 West Bank   
date_of_death                                                        2023-09-24   
gender                                                                        M   
took_part_in_the_hostilities                                                NaN   
place_of_residence                                               Nur Shams R.C.   
place_of_residence_district                                             Tulkarm   
type_of_injury                                                          gunfire   
ammunition                                                      live ammunition   
killed_by                                               Israeli security forces   
notes                         Fatally shot by Israeli forces while trying to...   

                                                                             2   \
name                                           'Abdallah 'Imad Sa'ed Abu Hassan   
date_of_event                                                        2023-09-22   
age                                                                        16.0   
citizenship                                                         Palestinian   
event_location                                                         Kfar Dan   
event_location_district                                                   Jenin   
event_location_region                                                 West Bank   
date_of_death                                                        2023-09-22   
gender                                                                        M   
took_part_in_the_hostilities                                                NaN   
place_of_residence                                                     al-Yamun   
place_of_residence_district                                               Jenin   
type_of_injury                                                          gunfire   
ammunition                                                      live ammunition   
killed_b

the following dataset has only one float variable (age) while all others are objects.

In [18]:
fatalities_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11124 entries, 0 to 11123
Data columns (total 16 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   name                          11124 non-null  object 
 1   date_of_event                 11124 non-null  object 
 2   age                           10995 non-null  float64
 3   citizenship                   11124 non-null  object 
 4   event_location                11124 non-null  object 
 5   event_location_district       11124 non-null  object 
 6   event_location_region         11124 non-null  object 
 7   date_of_death                 11124 non-null  object 
 8   gender                        11104 non-null  object 
 9   took_part_in_the_hostilities  9694 non-null   object 
 10  place_of_residence            11056 non-null  object 
 11  place_of_residence_district   11056 non-null  object 
 12  type_of_injury                10833 non-null  object 
 13  a

observing the size of the dataset allows one to know which variables have excessive null values.

In [19]:
fatalities_df.shape

(11124, 16)

In [9]:
fatalities_df.describe().T

count       mean        std  min   25%   50%   75%    max
age  10995.0  26.745703  13.780548  1.0  19.0  23.0  31.0  112.0

## Check NA values

in this case, the choice to remove rows with null values is wrong, since since since most of them are descriptive values, they do not affect the data statistically but only add information

In [16]:
print(sum(fatalities_df.isna().sum()))
print(fatalities_df.duplicated().sum())

7539
7


In [24]:
fatalities_df.isna().sum()

name                               0
date_of_event                      0
age                              129
citizenship                        0
event_location                     0
event_location_district            0
event_location_region              0
date_of_death                      0
gender                            20
took_part_in_the_hostilities    1430
place_of_residence                68
place_of_residence_district       68
type_of_injury                   291
ammunition                      5253
killed_by                          0
notes                            280
dtype: int64

let's check the percentage about this informations

In [36]:
print(f"percentage of age NA value: \t{(fatalities_df['age'].isna().sum()/len(fatalities_df))*100:.2f}%")
print(f"percentage of gender NA value: \t{(fatalities_df['gender'].isna().sum()/len(fatalities_df))*100:.2f}%")
print(f"percentage of took_part_in_the_hostilities NA value: \t{(fatalities_df['took_part_in_the_hostilities'].isna().sum()/len(fatalities_df))*100:.2f}%")
print(f"percentage of place_of_residence NA value: \t{(fatalities_df['place_of_residence'].isna().sum()/len(fatalities_df))*100:.2f}%")
print(f"percentage of place_of_residence_district NA value: \t{(fatalities_df['place_of_residence_district'].isna().sum()/len(fatalities_df))*100:.2f}%")
print(f"percentage of type_of_injury NA value: \t{(fatalities_df['type_of_injury'].isna().sum()/len(fatalities_df))*100:.2f}%")
print(f"percentage of ammunition NA value: \t{(fatalities_df['ammunition'].isna().sum()/len(fatalities_df))*100:.2f}%")
print(f"percentage of notes NA value: \t{(fatalities_df['notes'].isna().sum()/len(fatalities_df))*100:.2f}%")

percentage of age NA value: 	1.16%
percentage of gender NA value: 	0.18%
percentage of took_part_in_the_hostilities NA value: 	12.86%
percentage of place_of_residence NA value: 	0.61%
percentage of place_of_residence_district NA value: 	0.61%
percentage of type_of_injury NA value: 	2.62%
percentage of ammunition NA value: 	47.22%
percentage of notes NA value: 	2.52%


based on this information, the data will be rearranged as follows:
- Replace missing values for the variable age with the mean value (1.16%).
- Missing values for the variables sex, place_of_residence, place_of_residence_district, type_of_location, and notes are replaced with the mode value. this is so as not to change the distribution of the data too much.
- Remove the variables has_participated_in_hostilities, ammunition from the variable itself. The percentage of null values is excessive and would not lead to useful information for the entire population of the dataset.

In [37]:
age_mean = fatalities_df['age'].mean()
fatalities_df['age'].fillna(age_mean, inplace=True)

# **PART 2:** Exploratory Data Analysis